### PR TITLE
[TRRFFDA-513] Allow Conditional Rendering to support CDEs being used for multiple show/hides

### PR DESCRIPTION
**Pull request by alintreznai**

### CODE REVIEW

File: rdrf/rdrf/forms/dsl/parse_operations.py [hunk 1]
```diff
@@ -19,20 +19,28 @@ def invalid_cdes(self):
...
+            return self.target_cdes[0].cde
+        return None
+
     def __eq__(self, other):
         if isinstance(other, Target):
-            my_cdes = set(cde.cde for cde in self.target_cdes)
-            other_cdes = set(cde.cde for cde in other.target_cdes)
+            my_cdes = set(cde.get_key() for cde in self.target_cdes)
+            other_cdes = set(cde.get_key() for cde in other.target_cdes)
             return len(my_cdes - other_cdes) == 0
```
Comment (sztamas) [inline rdrf/rdrf/forms/dsl/parse_operations.py]: Why isn't this `my_cdes == other_cdes`?
What if `my_cdes` is a subset of `other_cdes`?

File: rdrf/rdrf/forms/dsl/parse_operations.py [hunk 2]
```diff
@@ -48,6 +56,9 @@ def __eq__(self, other):
     def __hash__(self):
         return hash((self.action, ))
```
Comment (sztamas) [inline rdrf/rdrf/forms/dsl/parse_operations.py]: Why isn't this `return hash(self.action)` instead?

**Review by sztamas — CHANGES_REQUESTED** [review]

File: rdrf/rdrf/forms/dsl/parse_operations.py (hunk 1 above)
Comment (alintreznai) [reply rdrf/rdrf/forms/dsl/parse_operations.py]: Updated to `self.target_cdes == other.target_cdes` and implemented `__eq__/__hash__ `for EnrichedCDE. They need to be equal so don't need to check for one being the subset of another.

File: rdrf/rdrf/forms/dsl/parse_operations.py (hunk 2 above)
Comment (alintreznai) [reply rdrf/rdrf/forms/dsl/parse_operations.py]: Indeed. Probably was a copy-paste from other `__hash__` implementations. Updated.

**Review by sztamas — APPROVED** [review]

**Pull request merged**